go/common/identity/identity.go

```diff
@@ -30,31 +32,124 @@ const (
 	tlsCertFilename = "tls_identity_cert.pem"
 )
 
+// ErrCertificateRotationForbidden is returned by RotateCertificates if
+// TLS certificate rotation is forbidden.  This happens when rotation is
+// enabled and an existing TLS certificate was successfully loaded
+// (or a new one was generated and persisted to disk).
+var ErrCertificateRotationForbidden = errors.New("identity", 1, "identity: TLS certificate rotation forbidden")
+
 // Identity is a node identity.
 type Identity struct {
+	sync.RWMutex
+
 	// NodeSigner is a node identity key signer.
 	NodeSigner signature.Signer
 	// P2PSigner is a node P2P link key signer.
 	P2PSigner signature.Signer
 	// ConsensusSigner is a node consensus key signer.
 	ConsensusSigner signature.Signer
-	// TLSSigner is a node TLS certificate signer.
-	TLSSigner signature.Signer
-	// TLSCertificate is a certificate that can be used for TLS.
-	TLSCertificate *tls.Certificate
+	// tlsSigner is a node TLS certificate signer.
+	tlsSigner signature.Signer
+	// tlsCertificate is a certificate that can be used for TLS.
+	tlsCertificate *tls.Certificate
+	// nextTLSCertificate is a certificate that can be used for TLS in the next rotation.
+	nextTLSCertificate *tls.Certificate
+	// DoNotRotate flag is true if we mustn't rotate the TLS certificates.
+	DoNotRotate bool
```

Comment:
Is the intention that this flag will also eventually encompass other signing keys?  If not, the name could be made more descriptive.

Reply:
Good point, I'll fix this.